// @y48er-ecom/common/build/events/base-listener.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.listener = void 0;
class listener {
    constructor(client) {
        this.ackWait = 5 * 1000;
        this.client = client;
    }
    subscriptionOptions() {
        return this.client
            .subscriptionOptions()
            .setDeliverAllAvailable()
            .setManualAckMode(true)
            .setAckWait(this.ackWait)
            .setDurableName(this.queueGroup);
    }
    listen() {
        const subscription = this.client.subscribe(this.subject, this.queueGroup, this.subscriptionOptions());
        subscription.on("message", (msg) => {
            console.log(`Message recieved ${this.subject} / ${this.queueGroup}`);
            const parseData = this.parseMessage(msg);
            this.onMessage(parseData, msg);
        });
    }
    parseMessage(msg) {
        const data = msg.getData();
        return typeof data === "string"
            ? JSON.parse(data)
            : JSON.parse(data.toString("utf-8"));
    }
}
exports.listener = listener;
